database/src/lib.rs

```diff
@@ -72,6 +72,175 @@
 
 use scylla::prepared_statement::PreparedStatement;
+use scylla::retry_policy::{QueryInfo, RetryDecision};
+use scylla::transport::errors::QueryError;
+
+#[derive(Debug)]
+pub struct CustomDBRetryPolicy {
+    max_retry: u8,
+    strict_mode: bool,
+}
+
+impl CustomDBRetryPolicy {
+    pub fn new(max_retry: u8, strict_mode: bool) -> CustomDBRetryPolicy {
+        CustomDBRetryPolicy {
+            max_retry,
+            strict_mode,
+        }
+    }
+}
+
+impl Default for CustomDBRetryPolicy {
+    /// By default using `strict_mode`
+    fn default() -> CustomDBRetryPolicy {
+        CustomDBRetryPolicy::new(0, true)
+    }
+}
+
+impl scylla::retry_policy::RetryPolicy for CustomDBRetryPolicy {
+    /// Called for each new query, starts a session of deciding about retries
+    fn new_session(&self) -> Box<dyn scylla::retry_policy::RetrySession> {
+        Box::new(CustomRetrySession::new(self.max_retry, self.strict_mode))
+    }
+
+    /// Used to clone this RetryPolicy
+    fn clone_boxed(&self) -> Box<dyn scylla::retry_policy::RetryPolicy> {
+        Box::new(CustomDBRetryPolicy::new(self.max_retry, self.strict_mode))
+    }
+}
+
+struct CustomRetrySession {
+    max_retry: u8,
+    max_retry_count: u8,
+    strict_mode: bool,
+}
+
+impl CustomRetrySession {
+    pub fn new(max_retry: u8, strict_mode: bool) -> CustomRetrySession {
+        CustomRetrySession {
+            max_retry,
+            max_retry_count: 0,
+            strict_mode,
+        }
+    }
+
+    /// Decide if we should retry the query
+    pub fn should_retry(&mut self) -> bool {
+        // If `strict_mode` always retry
+        if self.strict_mode {
+            return true;
+        };
+        // If `max_retry_count` is more than `max_retry` don't retry
+        if self.max_retry_count > self.max_retry {
+            return false;
+        };
+        // If `max_retry_count` is less than `max_retry` increment `max_retry_count` and retry
+        self.max_retry_count += 1;
+        true
+    }
+}
+
+impl Default for CustomRetrySession {
+    /// By default using `strict_mode`
+    fn default() -> CustomRetrySession {
+        CustomRetrySession::new(0, true)
+    }
+}
+
+impl scylla::retry_policy::RetrySession for CustomRetrySession {
+    /// Called after the query failed - decide what to do next
+    fn decide_should_retry(&mut self, query_info: QueryInfo) -> RetryDecision {
+        if let scylla::frame::types::LegacyConsistency::Serial(_) = query_info.consistency {
+            return RetryDecision::DontRetry;
+        };
+        tracing::warn!("ScyllaDB QueryError: {:?}", query_info.error);
+        match query_info.error {
+            QueryError::IoError(_)
+            | QueryError::DbError(scylla::transport::errors::DbError::Overloaded, _)
+            | QueryError::DbError(scylla::transport::errors::DbError::ServerError, _)
+            | QueryError::DbError(scylla::transport::errors::DbError::TruncateError, _) => {
+                tracing::info!(
```

Comment:
I believe it should be `warn` but not `info` for sure. This applies to all logs in this method